internal/sdkprovider/service/opensearch/opensearch_user.go

```diff
@@ -50,3 +57,84 @@ func ResourceOpenSearchUser() *schema.Resource {
 		Schema: aivenOpenSearchUserSchema,
 	}
 }
+
+// detourSecurityPluginEnabledCheck checks if the OpenSearch Security Plugin is enabled for the OpenSearch service.
+// If it is enabled, it returns an error, and the resource is not allowed to be created, read or updated.
+func detourSecurityPluginEnabledCheck(d *schema.ResourceData, m any) error {
+	client := m.(*aiven.Client)
+
+	project := d.Get("project").(string)
+	serviceName := d.Get("service_name").(string)
+
+	r, err := client.OpenSearchSecurityPluginHandler.Get(project, serviceName)
+	if err == nil && r.SecurityPluginAdminEnabled {
+		return errors.New("when the OpenSearch Security Plugin is enabled, OpenSearch users are being " +
+			"managed by it; delete the aiven_opensearch_user resource(s), and manage the users via the " +
+			"OpenSearch Security Plugin instead; any changes to the aiven_opensearch_user resource(s) are not " +
+			"going to have any effect now")
+	}
+
+	return err
+}
+
+// resourceOpenSearchUserCreate creates a OpenSearch User.
+func resourceOpenSearchUserCreate(ctx context.Context, d *schema.ResourceData, m any) diag.Diagnostics {
+	if err := detourSecurityPluginEnabledCheck(d, m); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return schemautil.ResourceServiceUserCreate(ctx, d, m)
+}
+
+// resourceOpenSearchUserRead reads a OpenSearch User into the Terraform state.
+func resourceOpenSearchUserRead(ctx context.Context, d *schema.ResourceData, m any) diag.Diagnostics {
+	_ = schemautil.ResourceServiceUserRead(ctx, d, m)
```

Comment:
Why do you ignore the error? Or should it go after "enabled" check?

Reply:
it is ignored because of the way OpenSearch Security Plugin works

Reply:
because of the way the OpenSearch Security Plugin works, once it's enabled, there will be an error when attempting to send this `GET` request to retrieve the error, and we want to prevent any errors from happening as we discussed earlier with the team; I will, however, update it now to add a validation for a specific error that should be skipped